src/Roslyn.Diagnostics.Analyzers/Core/AbstractDoNotCopyValue.cs

```diff
@@ -178,6 +238,140 @@ public void Dispose()
             }
         }
 
+        protected abstract class NonCopyableSymbolWalker : SymbolVisitor
+        {
+            private readonly SymbolAnalysisContext _context;
+            //private readonly HashSet<ISymbol> _handledSymbols = new();
+
+            protected NonCopyableSymbolWalker(SymbolAnalysisContext context, NonCopyableTypesCache cache)
+            {
+                _context = context;
+                Cache = cache;
+            }
+
+            protected NonCopyableTypesCache Cache { get; }
+
+            public sealed override void Visit(ISymbol? symbol)
+            {
+                base.Visit(symbol);
+            }
+
+            public override void DefaultVisit(ISymbol symbol)
+            {
+                base.DefaultVisit(symbol);
+            }
+
+            public override void VisitAlias(IAliasSymbol symbol)
+            {
+                base.VisitAlias(symbol);
+            }
```

Comment:
i don't quite get the purpose of the overrides.  is it just to be explicit that this is the behavior you want?  (if so, that's fine.  just trying to understand).

Reply:
Yes, this analyzer overrides everything.